### PR TITLE
Fix value template for global ComputeTargetHTTPSProxy

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Fixes b/345274481

Fix value template for global ComputeTargetHTTPSProxy. 
Resource created successfully:
```
Events:
  Type     Reason              Age              From                                Message
  ----     ------              ----             ----                                -------
  Warning  DependencyNotFound  49s              computetargethttpsproxy-controller  reference ComputeURLMap default/computetargethttpsproxy-dep-certmgr is not found
  Normal   Updating            16s              computetargethttpsproxy-controller  Update in progress
  Normal   UpToDate            4s (x2 over 5s)  computetargethttpsproxy-controller  The resource is up to date
```
### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by justinsb** [comment]
Looks good; is this covered by a test?

**Comment by maqiuyujoyce** [comment]
/lgtm

Agreed with @justinsb that we should add a test case for it.

**Comment by gemmahou** [comment]
Added dynamic test to cover `certificateMapRef` field

```
--- PASS: TestCreateNoChangeUpdateDelete (0.12s)
    --- PASS: TestCreateNoChangeUpdateDelete/compute (0.00s)
        --- PASS: TestCreateNoChangeUpdateDelete/compute/basic-globaltargethttpsproxy (301.47s)
PASS
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","msg":"Stopping and waiting for non leader election runnables"}
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","msg":"Stopping and waiting for leader election runnables"}
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","msg":"Stopping and waiting for caches"}
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","msg":"Stopping and waiting for webhooks"}
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","logger":"controller-runtime.webhook","msg":"Shutting down webhook server with timeout of 1 minute"}
{"severity":"info","timestamp":"2024-06-24T18:48:56.177Z","msg":"Wait completed, proceeding to shutdown the manager"}
ok      github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/dynamic      311.376s

```

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computetargethttpsproxy/globaltargethttpsproxy/create.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ spec:
   description: "test description"
   urlMapRef:
     name: computeurlmap-${uniqueId}
+  certificateMapRef:
```
Comment (maqiuyujoyce) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computetargethttpsproxy/globaltargethttpsproxy/create.yaml:23]: According to the field description: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/8d78e4962662663dbbb3de8262c6e12ed14e398f/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_computetargethttpsproxies.compute.cnrm.cloud.google.com.yaml#L64

`sslCertificates and certificateManagerCertificates fields can not be defined together.`

And according to the latest TF doc: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_target_https_proxy#certificate_manager_certificates

`Certificate manager certificates only apply when the load balancing scheme is set to INTERNAL_MANAGED. For EXTERNAL and EXTERNAL_MANAGED, use certificate_map instead.`

Feels like `certificateManagerCertificates` and `certificateMap` are mutually exclusive fields, and `certificateManagerCertificates`/ `certificateMap` and `sslCertificates` are mutually exclusive fields as well. I.e. only one out of the three fields can be configured. 

Could you verify that certificate map is indeed configured correctly in the underlying GCP resource when ssl certificates are configured?

Once the behavior is confirmed, could you update the field descriptions of those fields to reflect the correct usage of these fields?

**Comment by yuwenma** [comment]
/lgtm

Defer to @maqiuyujoyce comment on verifying the field and updating the doc.

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computetargethttpsproxy/globaltargethttpsproxy/create.yaml (hunk 1 above)
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computetargethttpsproxy/globaltargethttpsproxy/create.yaml]: The default value for `load balancing schemes` is EXTERNAL, so I'm able to created a target https proxy with `certificateMap`: https://screenshot.googleplex.com/7Lm35rNZ6hkmXMZ.png

However, `certificateManagerCertificates`/`certificateMap` and `sslCertificates` are mutually exclusive fields and I noticed that `sslCertificates` is discarded when both are set. I updated the test and field description.

**Comment by yuwenma** [comment]
/lgtm
/approve

Thank you for the change, looks good on my side

/hold In case @maqiuyujoyce want to take another pass.

File: config/servicemappings/compute.yaml [hunk 2]
```diff
@@ -2474,12 +2474,15 @@ spec:
           description: |-
             A reference to the CertificateMap resource uri that identifies a
             certificate map associated with the given target proxy. This field
-            can only be set for global target proxies.
+            can only be set for global target proxies. This field is only supported
+            for EXTERNAL and EXTERNAL_MANAGED load balancing schemes.
+            For INTERNAL_MANAGED, use certificate_manager_certificates instead.
```
Comment (maqiuyujoyce) [inline config/servicemappings/compute.yaml:2479]: Nit: we use field name in the format of camel case: `certificateManagerCertificates`.

File: config/servicemappings/compute.yaml [hunk 3]
```diff
@@ -2446,7 +2446,7 @@ spec:
             kind: CertificateManagerCertificate
             version: v1beta1
             group: certificatemanager.cnrm.cloud.google.com
-          valueTemplate: projects/{{project}}/locations/{{location}}/certificates/{{value}}
+          valueTemplate: projects/{{project}}/locations/global/certificates/{{value}}
```
Comment (maqiuyujoyce) [inline config/servicemappings/compute.yaml:2449]: I just realized you updated the valueTemplate for this reference: Could you also add a test case for it?

File: config/servicemappings/compute.yaml [hunk 4]
```diff
@@ -2496,7 +2499,7 @@ spec:
             kind: NetworkSecurityServerTLSPolicy
             version: v1beta1
             group: networksecurity.cnrm.cloud.google.com
-          valueTemplate: "projects/{{project}}/locations/{{location}}/serverTlsPolicies/{{value}}"
+          valueTemplate: "projects/{{project}}/locations/global/serverTlsPolicies/{{value}}"
```
Comment (maqiuyujoyce) [inline config/servicemappings/compute.yaml:2502]: Similarly, is there a test case covering this fix?

**Review by maqiuyujoyce — COMMENTED** [review]
Looks like there are fixes for three fields. Wondering if you can add test cases for the other two fixes.

File: config/servicemappings/compute.yaml (hunk 3 above)
Comment (gemmahou) [reply config/servicemappings/compute.yaml]: added test

File: config/servicemappings/compute.yaml (hunk 4 above)
Comment (gemmahou) [reply config/servicemappings/compute.yaml]: I noticed another issue with serverTlsPolicyRef. There's issue when resolving `project`:   `"error": "Update call failed: error fetching live state: error converting resource config: unable to resolve missing value: project"`, and I already specified `projectRef` in the yaml for NetworkSecurityServerTLSPolicy resource. 

This field is introduced in #941, and this is a DCL based reference field. I don't think it's necessary to address it here, so I reverted my change back and will create a ticket for this. Lmk if that sounds reasonable.

**Pull request merged**